### DISCUSSION TITLE
Delete key does not work

### DISCUSSION THREAD

**Discussion by bukshee** [helix-editor/helix#7563] [Troubleshooting]
Hi,

In Helix in Insert mode neither Delete nor Alt-Delete work. They should:
https://docs.helix-editor.com/keymap.html#insert-mode

Ctrl+d and Alt-d do work though. Even if I add keymap in conf:
[keys.insert]
"A-del" = "delete_word_forward"

If I exit helix back to the terminal (st) both work fine. Except inside helix.

Any ideas? Thank You!

helix 23.05
Manjaro linux.

**Comment by rojebd** [comment]
@bukshee 
same issue have you found a solution yet?

**Comment by TornaxO7** [comment]
> helix 23.05

Maybe try that out with the latest version of helix (23.10)

**Reply by rojebd** [reply]
i am on the latest

**Comment by kirawi** [comment]
I'm going to guess that you need a terminal that supports the enhanced keyboard protocol for those keybinds. `st` also seems to have some issues with the delete key based on the immediate results of a google search.

**Reply by kirawi** [reply]
I checked `st` and I can replicate it with that terminal. I'm not interested in investigating further (it's likely something on st's side), but I would suggest using something else instead.

**Comment by babucarr32** [comment]
If you are  on macOs

delete = `backspace`
fn + delete = `del`

`delete` deletes backwards
`fn + delete` deletes forwards
